be/tool/beck.c

```diff
@@ -511,6 +526,31 @@ int main(int argc, char **argv)
 		      err(EX_NOINPUT, "Cannot seek snapshot to the beginning.");
 		printf("Snapshot size: %"PRId64".\n", s.s_size);
 	}
+
+	generation_id_get(s.s_file, &gen_id);
+	if (print_gen_id) {
```

Comment:
What if gen_id is zero here and s.s_gen is not equal to zero , it shouldn't return from here?

Reply:
Well if we have (gen_id == 0) and (s.s_gen == 0) then we will return since we are not sure which is the latest Generation ID to use. We might find the latest one by scanning the whole MD and then providing it somehow from the command-line, but we would do that in next version as that would need changes in 2 scripts to accept the parameter and given the time crunch we postponed it for later.